lib/yi_appium_caps_util.rb

```diff
@@ -55,15 +55,16 @@ def getTeamID()
       end
     end
 
-    def run_update(parsed_data)
+    def run_update(parsed_data, platformName_value: nil)
       #Make a copy of the parsed data
       output_data = Marshal.load(Marshal.dump(parsed_data))
 
       if parsed_data['caps'] == nil
         raise '[caps] is missing form appium.txt'
       else
-        platformName_value = parsed_data['caps']['platformName']
-
+        if platformName_value == nil
+          platformName_value = parsed_data['caps']['platformName']
+        end
         if platformName_value == nil
```

Comment:
i'm not sure if I got this correct, is `if platformName_value == nil` repeated?

Reply:
if no `platformName_value` is passed, try to get it from the caps.
if you can get it from caps, then raise an exception

Reply:
Yeaa I see what we're trying to achieve here.. I just feel there is a neater way to do it in ruby by leveraging `unless` statements. So maybe something along the lines

```
      raise '[caps] is missing form appium.txt' unless parsed_data['caps'] != nil
      if platformName_value == nil then platformName_value = parsed_data['caps']['platformName']
      raise 'platformName is missing from appium.txt' unless platformName_value != nil

      case platformName_value.downcase
      when 'android'
        update_android_caps (output_data)
      when 'ios'
        update_ios_caps (output_data)
      else
        raise 'platformName: ' + platformName_value + ' is not supported'
      end

      return output_data
```

But other than that, logic looks good